File: cpp/LeetCode3403.cpp
class Solution {
    public:
        string answerString(string word, int numFriends) {
            if (numFriends == 1) {
                return word;
            }
            
            int max_len = word.length() - numFriends + 1;
            string maxSubword = "";
            
            // Check all the word with the maximum len
            for (int i = 0; i < numFriends; i++) {
                string w = word.substr(i, max_len);
                if (w > maxSubword) {
                    maxSubword = w;
                }
            }
            
            // Check words with not maximum len that are at the end of the string
            for (int i = 1; i < max_len; i++) {
                string w = word.substr(word.length() - i);
                if (w > maxSubword) {
                    maxSubword = w;
                }
            }
            
            return maxSubword;
        }
    };

File: python/LeetCode3403.py
class Solution:
    def answerString(self, word: str, numFriends: int) -> str:
        if numFriends == 1:
            return word

        max_len = len(word) - numFriends + 1
        maxSubword = ""

        # Check all the word with the maximum len
        for i in range(numFriends):
            w = word[i : i + max_len]
            if w > maxSubword:
                maxSubword = w

        # Check words with not maximum len that are at the end of the string
        for i in range(1, max_len):
            w = word[-i:]
            if w > maxSubword:
                maxSubword = w

        return maxSubword
